File: Solutions/1260. Shift 2D Grid/1260. Shift 2D Grid.go
func shiftGrid(grid [][]int, k int) [][]int {
    m,n:=len(grid),len(grid[0])
    st:=m*n-k%(m*n)
    var res [][]int
    for i:=st;i<st+m*n;i+=1{
        j:=i%(m*n)
        r,c:=j/n,j%n
        if (i-st)%n==0{
            res=append(res,[]int{})
        }
        res[len(res)-1]=append(res[len(res)-1],grid[r][c])
    }
    return res
}

File: Solutions/1260. Shift 2D Grid/1260. Shift 2D Grid.cpp
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        int m=grid.size(),n=grid[0].size();
        int st=m*n-k%(m*n);
        vector<vector<int>> res;
        for(int i=st;i<m*n+st;++i){
            int j=i%(m*n),r=j/n,c=j%n;
            if((i-st)%n==0){
                res.push_back({});
            }
            res.back().push_back(grid[r][c]);
        }
        return res;
    }
};